consumer/src/mode/decoded/deposited.rs

```diff
@@ -520,16 +532,12 @@ impl Deposited {
         )
         .await?
         {
-            Some(mut pos) => {
-                pos.shares = U256Wrapper::from(self.receiverTotalSharesInVault);
-                pos
+            Some(mut position) => {
+                position.shares =
+                    position.shares + U256Wrapper::from(self.receiverTotalSharesInVault);
```

Comment:
if a position exists already, we need to update the shares by summing it up, not override it